src/actions/analysis_queue.rs

```diff
@@ -335,14 +335,14 @@ impl IsolatedAnalysisJob {
                 } else {
                     self.context.clone()
                 };
+                self.report.send(TimestampedStorage::make_isolated_result(
+                    self.timestamp,
+                    analysis.clone())).ok();
                 self.notify.send(ServerToHandle::IsolatedAnalysisDone(
                     self.path.clone(),
                     new_context,
                     analysis.get_import_names()
                 )).ok();
-                self.report.send(TimestampedStorage::make_isolated_result(
-                    self.timestamp,
-                    analysis)).ok();
             },
```

Comment:
@JonatanWaern we noticed a race condition here for our suppressed imports mode.
On more than half of the runs, we would get an error "Failed to get isolated analysis to trigger LinterJob" and the DFA client would not close afterwards.

We noticed it was caused by message IsolatedAnalysisDone sent to server before the make_isolated_result could store the actual analysis in the analysis_storage.

We inverted the sequence for sending these two messages to resolve that. I consider this should be safer in general.